lightning/src/util/events.rs

```diff
@@ -319,15 +328,18 @@ impl MaybeReadable for Event {
 					let mut payment_hash = PaymentHash([0; 32]);
 					let mut rejected_by_dest = false;
 					let mut network_update = None;
+					let mut all_paths_failed = true;
 					read_tlv_fields!(reader, {
 						(0, payment_hash, required),
 						(1, network_update, ignorable),
 						(2, rejected_by_dest, required),
+						(3, all_paths_failed, required),
```

Comment:
`required` implies that we'll fail to read old versions that are missing the field. I think we should just initialize it to `Some(true)` and unwrap it below, keeping the broken behavior for events that were pending.

Reply:
Ah, I thought if `all_paths_failed` was an `option` `fieldty` in `read_tlv_fields`, it had to be an `option` in `write_tlv_fields` also. Updating mental model since they def can be asymmetric.

Sorry, I finally drew out the 2x2 tlv ser matrix (`type` odd/even is for old clients reading new, `fieldty` option/required is for new clients reading old)

Reply:
Right, everything is written the same whether its an option or not, option just declares whether we require it to be present or not, and odd/even declares whether old versions that don't understand it will fail or ignore.